### DISCUSSION TITLE
Directional NFFT

### DISCUSSION THREAD

**Discussion by tknopp** [JuliaMath/NFFT.jl#66] [Show and tell]
Hey @robertdj,

I just generalized the directional NFFT that you implemented in a way that one can also do 2D or ND NFFTs within a larger-dimensional tensor.

This made me wonder:
- could you point me to some paper where the directional NFFT is actually discussed / used?
- are you aware of any application where my generalization could be useful?

In my opinion our implementation is only beneficial if the NFFT is not applied over the most inner dimension because in that case the approach with slicing would be probably equally fast.

Thanks!

Tobi

**Comment by robertdj** [comment]
Hi @tknopp 

I don't have any references, sorry. I used it in my package https://github.com/robertdj/GeneralizedSampling.jl that I have unfortunately not touched in a loong time. There it is used for both rows and columns of matrices.

As I recall, slicing was not available at that time (or at least not as slick as the current `@view` macro :-))

Best,

Robert

**Reply by tknopp** [reply]
I tried it with views but that is much slower. If I get it right that is because the order of the for loops matters.

**Comment by roflmaostc** [comment]
I think the application makes sense in the context of CUDA and if you got a batch dimension.
Doing the for loop along this batch dimension is slower than if we would parallelize those slices with CUDA.
